### DISCUSSION TITLE
Multiple #[Rules()] fail when including custom error messages.

### DISCUSSION THREAD

**Discussion by citricguy** [livewire/livewire#6187] [7. Bugs (archived)] [answered]
### Livewire version

3.0.0 Beta.7

### Which PHP version are you using?

PHP 8.2

### Repository URL

https://github.com/citricguy/6187-bug

### Steps To Reproduce

To reproduce add any two rules to a variable that have an attached custom message and then attempt validation. 

I added a repo with an example and a test but everything pertinent is copied and pasted below.

- Example repo:  https://github.com/citricguy/6187-bug.git

- type foo in the field and click save (to fail min:5 rule)
- exception is thrown.

Multiple Rules work now after updating to beta7, however, if the rules have custom messages validation will fail.

Here is the validation I'm using (Full file's shown below)
```
    #[Rule('required', message: 'Please enter a title.')]
    #[Rule('min:5', message: 'Your title is too short.')]
```

When an empty string is supplied (failing 'required) I get:
`htmlspecialchars(): Argument #1 ($string) must be of type string, array given`

When a 2 character string is supplied (failing 'min:5') I get:
`Undefined array key "string"`

I'll share the code I used [from the documentation](https://livewire.laravel.com/docs/validation#custom-messages) for use in re-producing the error:

Here is RuleBug.php

```
namespace App\Livewire;

use Livewire\Attributes\Rule;
use Livewire\Component;

class RuleBug extends Component
{

    #[Rule('required', message: 'Please enter a title.')]
    #[Rule('min:5', message: 'Your title is too short.')]
    public $title = '';

    public function save()
    {
        $this->validate();
    }
    public function render()
    {
        return view('livewire.rule-bug');
    }
}

```

Here is rule-bug.blade.php
```
<form wire:submit="save">
    <input type="text" wire:model="title">
    <div>
        @error('title') <span class="error">{{ $message }}</span> @enderror
    </div>
    <button type="submit">Save</button>
</form>
```

Here is the route:

`Route::get('/', \App\Livewire\RuleBug::class)->name('rulebug');`

Finally here is the test:

```
<?php

namespace Tests\Feature;

use App\Livewire\RuleBug;
use Livewire\Livewire;
use Tests\TestCase;

class BugTest extends TestCase
{
    /**
     * A basic feature test example.
     */
    public function test_example(): void
    {
        Livewire::test(RuleBug::class)
            ->set('title', 'foo')
            ->call('save')
            ->assertNoErrors(['title' => 'min']);
    }
}
```

When custom error messages are not used, multiple rules seem to work just fine. i.e.,

```
use Livewire\Attributes\Rule;
 
#[Rule('required')]
#[Rule('min:5')]
public $title = '';
```

### Additional information

I've been staring at this for half an hour, I'm hoping it's user-error. 😁

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment, marked as answer]
@citricguy thanks for reporting! A failing test PR #6217 has already been submitted which I think covers this issue.

**Reply by citricguy** [reply]
This issue seems to be resolved in the most recent beta. Thank you.
